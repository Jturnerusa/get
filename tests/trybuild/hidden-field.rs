use get::Get;

#[derive(Get)]
struct Crab {
    #[get(hide)]
    name: String,
    age: u64,
}

fn crab() {
    let ferris = Crab {
        name: "ferris".to_string(),
        age: 1,
    };
    assert_eq!(ferris.name().as_str(), "ferris");
}

pub fn main() {}
