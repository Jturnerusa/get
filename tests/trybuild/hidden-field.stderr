error: expected identifier, found `1`
  --> tests/trybuild/hidden-field.rs:13:9
   |
11 |     let ferris = Crab {
   |                  ---- while parsing this struct
12 |         name: "ferris".to_string(),
13 |         1
   |         ^ expected identifier

error[E0063]: missing field `age` in initializer of `Crab`
  --> tests/trybuild/hidden-field.rs:11:18
   |
11 |     let ferris = Crab {
   |                  ^^^^ missing `age`

error[E0599]: no method named `name` found for struct `Crab` in the current scope
  --> tests/trybuild/hidden-field.rs:15:23
   |
4  | struct Crab {
   | ----------- method `name` not found for this struct
...
15 |     assert_eq!(ferris.name().as_str(), "ferris");
   |                       ^^^^-- help: remove the arguments
   |                       |
   |                       field, not a method
